### PR TITLE
docs: adding ui libraries intergration docs and example

### PR DESCRIPTION
**Pull request by enyelsequeira**
# Changelog

- Hello, I have added intergation docs with other UI libraries, the ones I've used are `Material-ui` and `Mantine` 

in regards to this 
https://github.com/TanStack/form/issues/473

Thank you

### CODE REVIEW

**Comment by fulopkovacs** [comment]
One note: you'll have to update the `config.json` file if you want to add this page to the _"Guides"_ group in the sidebar:
```json
    {
      "label": "Guides",
      "children": [
      ...,
        {
          "label": "UI Libraries",
          "to": "guides/ui-libraries"
        }
      ]
    }
```

This is what the docs site's sidebar looks like after the `config.json` gets updated (and the branch is synced up with `TanStack/form:main`):
image

File: docs/guides/ui-libraries.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          />
+
+```
+
+- The integration approach is the same as with Mantine.
+- The primary difference lies in the specific Material-UI component properties and styling options.
+
+# Sandbox example with `Material` and `Mantine`
+
+[UI Library intergration](https://codesandbox.io/p/devbox/demo-form-query-x7w2gz)
```
Comment (crutchcorn) [inline docs/guides/ui-libraries.md:132]: I don't think this should be needed. Instead, make a directory called "ui-libraries" in examples in this repo.

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Looks good! Minor feedback on top of the required config changes. 

Also, with the config make sure to put this under React, not Core

File: docs/guides/ui-libraries.md (hunk 1 above)
Comment (enyelsequeira) [reply docs/guides/ui-libraries.md]: Alright done

**Comment by enyelsequeira** [comment]
Hey @crutchcorn  could you review this, i've added a guide folder under react, since I am assuming some docs specifically for react will be in here, and also I've tried adding my example setup of the sandbox, but couldn't get it to work, let me know if you think there are changes needed

Thank you!

**Review by crutchcorn — APPROVED** [review]
Great work on this!

**Comment by enyelsequeira** [comment]
Hey @crutchcorn have you seen this, not sure since it wasn't merged, missing something?

**Comment by crutchcorn** [comment]
Sorry @enyelsequeira! You're golden, nothing missing on your end. I was trying to fix our build pipeline which was causing this PRs build errors and then got nerd sniped into working on `@tanstack/config`. 

Should try to merge this today or early this week.

**Comment by crutchcorn** [comment]
@enyelsequeira can you merge in changes from `main` to this branch so CI/CD passes? :)

**Pull request merged**